subsys/net/l2/ieee802154/ieee802154_mgmt.c

```diff
@@ -574,7 +574,31 @@ static int ieee802154_associate(uint64_t mgmt_request, struct net_if *iface,
 	 * TODO: The Association Response command shall be sent to the device
 	 *       requesting association using indirect transmission.
 	 */
-	k_sem_take(&ctx->scan_ctx_lock, K_USEC(ieee802154_get_response_wait_time_us(iface)));
+	ret = k_sem_take(&ctx->scan_ctx_lock, K_USEC(ieee802154_get_response_wait_time_us(iface)));
+
+	if (ret == -EAGAIN) {
+		/* Timeout, send DATA_REQUEST */
+		net_pkt_unref(pkt);
+		pkt = ieee802154_create_mac_cmd_frame(
+				iface, IEEE802154_CFI_DATA_REQUEST, &params);
+		if (!pkt) {
```

Comment:
Please change this to `if (pkt == NULL) {` as dictated by MISRA so that we do not introduce new code that is using boolean checks for non boolean variables.